packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -64,35 +71,47 @@ const KeyboardTypeToNativeValuesMap = {
   decimal: 'decimal-pad',
 };
 
-const StyledNativeBaseInput = styled.TextInput<
-  Omit<
-    StyledBaseInputProps,
-    'accessibilityProps' | 'setCurrentInteraction' | 'currentInteraction'
-  > & {
-    isTextArea?: boolean;
-    isFocused: boolean;
-    autoCompleteType?: typeof autoCompleteSuggestionTypeAndroid[keyof typeof autoCompleteSuggestionTypeAndroid];
-  }
->((props) => ({
-  ...getBaseInputStyles({
-    theme: props.theme,
-    isFocused: props.isFocused,
-    isDisabled: !props.editable,
-    validationState: props.validationState,
-    leadingIcon: props.leadingIcon,
-    prefix: props.prefix,
-    interactionElement: props.interactionElement,
-    suffix: props.suffix,
-    trailingIcon: props.trailingIcon,
-  }),
-  lineHeight: undefined,
-  textAlignVertical: 'top',
-  height: props.isTextArea
-    ? `${props.theme.typography.lineHeights.xl * (props.numberOfLines ?? 0)}px`
-    : '36px',
-}));
+type StyledComponentInputProps = Omit<
+  StyledBaseInputProps,
+  'accessibilityProps' | 'setCurrentInteraction' | 'currentInteraction'
+> & {
+  isTextArea?: boolean;
+  isFocused: boolean;
+  autoCompleteType?: typeof autoCompleteSuggestionTypeAndroid[keyof typeof autoCompleteSuggestionTypeAndroid];
+};
+
+const getRNInputStyles = (
+  props: StyledComponentInputProps &
+    ThemeProps<DefaultTheme> &
+    (TextInputProps | TouchableHighlightProps),
+): CSSObject => {
+  return {
+    ...getBaseInputStyles({
+      theme: props.theme,
+      isFocused: props.isFocused,
+      isDisabled: 'editable' in props ? !props.editable : undefined,
+      validationState: props.validationState,
+      leadingIcon: props.leadingIcon,
+      prefix: props.prefix,
+      interactionElement: props.interactionElement,
+      suffix: props.suffix,
+      trailingIcon: props.trailingIcon,
+    }),
+    lineHeight: undefined,
+    textAlignVertical: 'top',
+    height: props.isTextArea
+      ? `${props.theme.typography.lineHeights.xl * (props.numberOfLines ?? 0)}px`
+      : '36px',
```

Comment:
Might actually be better to have a TODO comment here to fix this magic number later when we have size tokens.